app/src/main/java/com/simplemobiletools/calculator/activities/MainActivity.kt

```diff
@@ -0,0 +1,167 @@
+package com.simplemobiletools.calculator.activities
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.TextView
+
+import butterknife.BindView
+import butterknife.ButterKnife
+import butterknife.OnClick
+import butterknife.OnLongClick
+import com.simplemobiletools.calculator.*
+import me.grantland.widget.AutofitHelper
+
+class MainActivity : SimpleActivity(), Calculator {
+    @BindView(R.id.result) lateinit var result: TextView
+    @BindView(R.id.formula) lateinit var formula: TextView
```

Comment:
these wont be needed after removing butterknife, just do "result.setOnClickListener {}" etc wherever needed

Reply:
Should I instantiate views with `val result by lazy { findViewById<TextView>(R.id.result) }` or something?

Reply:
no, dont store views in any variable. Just do "result.setOnClicklistener {" ... or whatever where needed

Reply:
down in MainActivity instead of "value = result.text.toString().trim { it <= ' ' }" you can use "value = result.value", it will automatically get any textview/edittext text and trim it

Reply:
From what I see `result` has nothing to do with clicks. Can't realize the idea...

Reply:
ye it doesnt need a clicklistener, I wanted to say that you just call whatever function you need directly on "result", no need for any findViewById or so